### DISCUSSION TITLE
Programmatically change center

### DISCUSSION THREAD

**Discussion by bradley-varol** [giorgiabosello/google-maps-react-markers#40] [Help] [answered]
### Describe the feature you'd like to request

In `google-map-react`, we can use the `center` prop to programmatically change the center of the map (e.g. center when clicking on a marker).

How do we navigate the map to a lat/lng?

### Describe the solution you'd like

-

### Describe alternatives you've considered

-

**Comment by giorgiabosello** [comment, marked as answer]
You need to use the methods inside the mapRef to change the map after being initialized.
Example:
```js
const mapRef = useRef(null)

const onGoogleApiLoaded = ({ map }) => {
    mapRef.current = map
    mapRef.current.setCenter(new lat/lng)
}

return (
  <GoogleMap
    // options
    onGoogleApiLoaded={onGoogleApiLoaded}
  ></GoogleMap>
```

**Reply by bradley-varol** [reply]
Awesome! Thank you for such a quick response 🙏🏻

**Reply by bradley-varol** [reply]
Is the LatLng class available from this package? e.g. GoogleMapReact.LatLng? I have no autocomplete using this package 🤔

**Reply by giorgiabosello** [reply]
@bradley-varol nope, but you have access to all the methods of the Google Maps library with the ref and it should be there, somewhere.

See [here](https://developers.google.com/maps/documentation/javascript/examples/map-latlng-literal).
